presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -771,6 +771,42 @@ public void testSchemaAuthorization()
         assertUpdate(admin, "DROP ROLE admin");
     }
 
+    @Test
+    public void testTableAuthorization()
+    {
+        Session admin = Session.builder(getQueryRunner().getDefaultSession())
+                .setCatalog(getSession().getCatalog().get())
+                .setIdentity(Identity.forUser("hive").withRole("hive", new SelectedRole(ROLE, Optional.of("admin"))).build())
+                .build();
+
+        Session alice = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setIdentity(Identity.forUser("alice").withPrincipal(getSession().getIdentity().getPrincipal()).build())
+                .build();
+
+        Session bob = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setIdentity(Identity.forUser("bob").withPrincipal(getSession().getIdentity().getPrincipal()).build())
+                .build();
+
+        assertUpdate(admin, "CREATE SCHEMA test_table_authorization");
+        assertUpdate(admin, "CREATE ROLE admin");
+
+        assertUpdate(admin, "CREATE TABLE test_table_authorization.foo (col int)");
+        assertUpdate(admin, "ALTER TABLE test_table_authorization.foo SET AUTHORIZATION alice");
+        assertQueryFails(bob,
+                "ALTER TABLE test_table_authorization.foo SET AUTHORIZATION alice",
+                "Access Denied: Cannot set authorization for table test_table_authorization.foo to USER alice");
+        assertUpdate(alice, "ALTER TABLE test_table_authorization.foo SET AUTHORIZATION bob");
```

Comment:
Now it would be nice to have `SHOW CREATE TABLE` to return information who is the owner. Then we could use it for tests.

Actually, having `CREATE TABLE (...) WITH AUTHORIZATION 'bob'` would be a good feature. No need to do two calls `CREATE` and `ALTER`. Same for SCHEMA. Let's capture this as github issue, for further discussion.

Reply:
This is the next thing to do. Captured as https://github.com/prestosql/presto/issues/5722